### PR TITLE
DAOstar Collab: Improve ProposalsURI - Publish more info for proposals

### PR DESCRIPTION
**Pull request by Rashmi-278**
Fixes # .

Changes proposed in this pull request:
- Add Content URI Field, which includes Author, IPFS hash, Body, Link of the proposal
- Add Discussion URI Field, which includes the forum discussion link

Reference Issues:
- https://github.com/metagov/daostar/issues/203
- https://github.com/metagov/daostar/issues/239

### CODE REVIEW

File: src/eip4824.ts [hunk 1]
```diff
@@ -82,6 +82,13 @@ router.get('/:space/proposals', async (req, res) => {
     type: 'proposal',
     id: proposal.id,
     name: proposal.title,
+    contentURI: {
+      body: proposal.body,
+      author: proposal.author,
+      ipfs: proposal.ipfs,
+      link: proposal.link
+    },
+    discussionURI: proposal.discussion,
```
Comment (ChaituVR) [inline src/eip4824.ts:91]: I get `"contentURI": {},` and no `discussionURI` field in response

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/eip4824.ts (hunk 1 above)
Comment (Rashmi-278) [reply src/eip4824.ts:91]: It should be fixed now, can you check?

File: src/eip4824.ts [hunk 2]
```diff
@@ -95,6 +102,7 @@ router.get('/:space/proposals', async (req, res) => {
   });
 });
 
+
```
Comment (ChaituVR) [inline src/eip4824.ts:105]:
```suggestion
```

File: src/eip4824.ts [hunk 3]
```diff
@@ -82,6 +82,13 @@ router.get('/:space/proposals', async (req, res) => {
     type: 'proposal',
     id: proposal.id,
     name: proposal.title,
+    contentURI: {
+      body: proposal.body,
+      author: proposal.author,
+      ipfs: proposal.ipfs,
+      link: proposal.link
```
Comment (ChaituVR) [inline src/eip4824.ts:89]: There is no link field on proposal 🤔 what should be the value

Comment (bonustrack) [reply src/eip4824.ts:89]: The snapshot URL of the proposal, we have this on the hub API

Comment (Rashmi-278) [reply src/eip4824.ts:89]: Link of the Snapshot Proposal

`"link": "https://snapshot.org/#/opcollective.eth/proposal/0xc9c0a739a29ce6477708f49a2985e4c8cb2e4ba0be1d3761541ed189a9c341e1",`

I found this from GraphQL API,

Comment (ChaituVR) [reply src/eip4824.ts:89]: @Rashmi-278 no access for me to change it,

Could you change these:
- Remove `link` from the above query
- You can use the same logic as https://github.com/metagov/snapshot-hub/blob/pr/Rashmi-278/911/src/graphql/helpers.ts#L401-L402

Comment (Rashmi-278) [reply src/eip4824.ts:89]: I face 404 on the above link, do you mean to remove link from query and `Query *` ?

File: src/eip4824.ts [hunk 4]
```diff
@@ -70,7 +70,7 @@ router.get('/:space/proposals', async (req, res) => {
     if (!space.verified) return res.status(400).json({ error: 'INVALID' });
 
     proposals = await db.queryAsync(
-      'SELECT id, title, start, end FROM proposals WHERE space = ? ORDER BY created DESC LIMIT 20',
+      'SELECT id, title, start, end, body, author, ipfs, link, discussion FROM proposals WHERE space = ? ORDER BY created DESC LIMIT 20',
```
Comment (ChaituVR) [inline src/eip4824.ts:73]:
```suggestion
      'SELECT id, title, start, end, body, author, ipfs, discussion FROM proposals WHERE space = ? ORDER BY created DESC LIMIT 20',
```

**Pull request merged**